src/tests/Interop/PInvoke/IEnumerator/IEnumeratorNative.cpp

```diff
@@ -98,6 +98,13 @@ extern "C" DLL_EXPORT HRESULT STDMETHODCALLTYPE VerifyIntegerEnumeration(IDispat
         return hr;
     }
 
+    hr = VariantClear(&result);
```

Comment:
This should be immediately after the `QueryInterface()`. The reason being the lifetime of the `VARIANT` is transferred in the QI, so it can be released at that point, regardless of success or failure of that QI.